File: tell-dont-ask-kata-kotlin/src/main/java/it/gabrieletondi/telldontaskkata/domain/Order.kt
package main.java.it.gabrieletondi.telldontaskkata.domain

import java.math.BigDecimal

class Order {
    var total: BigDecimal? = null
    var currency: String? = null
    private var items: MutableList<OrderItem>? = null
    var tax: BigDecimal? = null
    var status: OrderStatus? = null
    var id: Int = 0

    fun getItems(): List<OrderItem>? {
        return items
    }

    fun setItems(items: MutableList<OrderItem>) {
        this.items = items
    }

    fun addItem(orderItem: OrderItem) {
        items!!.add(orderItem)
        total = total!!.add(orderItem.taxedAmount!!)
        tax = tax!!.add(orderItem.tax!!)
    }
}

File: tell-dont-ask-kata-ruby/models/Order.rb
class Order
    attr_accessor :total, :currency, :items, :tax, :status, :id

    def initialize(total: 0.00, currency: 'Unknown', items: [], tax: 0.00, status: OrderStatus::CREATED, id: nil)
        @total = total
        @currency = currency
        @items = items
        @tax = tax
        @status = status
        @id = id
    end

end
